src/main/java/org/kohsuke/github/GHCheckRun.java

```diff
@@ -21,8 +21,18 @@
     private String conclusion;
     private String name;
     private String headSha;
+    private String nodeId;
+    private String externalId;
+    private String startedAt;
+    private String completedAt;
```

Comment:
I just saw the `GHObject` uses `String` for such timestamps at first and convert them into `Date` in getters. But yes, using `Date` from the beginning is better. I've changed it.

Reply:
> > Just so I understand it: the reason why there aren't more tests in here is because, for general reading of check runs (outside of events) we need to have a `GHCheckSuite` for them right?
> 
> I'm not sure which kinds of tests our project requires, and I normally do not test the getters and setters, but I assume we have to test them here because the values are bound by Jackson? Also, I'm not sure how complete our tests have to be, for example, there is a `GHRepository`in our `GHEventPayload.CheckRun`, do we have to test whether each value of `GHRepository`is correct? I think such a recursive style is not necessary, so I just test some basic properties like `id` or `name` of the objects in `CheckRun` to make sure that Jackson reaches those.

I'm going to give you vague-ish answer.  We require test coverage for any new code/features added to the project, but that is a recent decision, so there are lots of gaps  in test coverage for the project overall.  You do not need to do recursive tests, but we do need tests that show the objects are behaving correctly overall.  You can save yourself some testing by not having public setters - then you only need to touch each of the getters at least once. Write a set of tests that your feel are reasonable and we can iterate to add more if they seem needed.